cmd/check/main.go

```diff
@@ -32,12 +33,27 @@ func main() {
 		log.Fatalf("failed unmarshalling: %s", err)
 	}
 
-	client := boshio.NewClient(nil, nil)
+	client := boshio.NewClient(nil, nil, checkRequest.Source.ForceRegular)
 	stemcells, err := client.GetStemcells(checkRequest.Source.Name)
 	if err != nil {
 		log.Fatalf("failed getting stemcell: %s", err)
 	}
 
+	supportsLight := client.SupportsLight(stemcells)
+	if supportsLight {
+		if checkRequest.Source.ForceRegular {
+			regularFilter := func(stemcell boshio.Stemcell) bool {
+				return stemcell.Regular != nil
+			}
+			stemcells = client.FilterStemcells(regularFilter, stemcells)
+		} else {
+			lightFilter := func(stemcell boshio.Stemcell) bool {
+				return stemcell.Light != nil
```

Comment:
@vito the `supportsLight` returns true if that IaaS supports light stemcells, so we don't execute the `ForceRegular` check for IaaS that only have regular stemcells.

Cases:
- IaaS only supports regular, don't both filtering
- IaaS supports light and ForceRegular is true, only return Regular stemcells
  - This is used in our light stemcell builder pipeline to only fetch heavy stemcells
- IaaS supports light and ForceRegular is false, only return Light stemcells
  - This ensures the resource waits until the light version is out before running. This is important as the heavy GCE stemcells will be available an hour or so before the light versions.